botpress/botpress
# chore(dev): new lint rules & dx changes

I created a new PR since I changed the scope of my previous one (no more pre-commit hooks).

- The new TSLint rules will be checked by the CI for your current changes once the PR is opened
- Use `yarn run cmd lint` or `yarn run cmd lint --fix` to test/fix your staged files against the new rules
- Added a hook that will prevent committing files larger than 30mb (only user-side - not enforced)
- Removed the specific `tslint.json` file that were in the `views` folder (just fixed the issue in the main tslint file)
- Removed the rule preventing the use of `null` 
- Updated to the latest version of typescript
- Started moving config files from the main folder to the `config` folder

Changes to ui-admin were made quickly because the new version of typescript didn't like the code.

Here's a summary of the new rules that will be checked by the linter: 

- Enforce ES6 shorthand notation for methods and property
- Use an interface instead of a type when a type is not required
- Avoid using backticks unless you really need it for a template literal
- Use string interpolation instead of concatenating string (ex: "text" + 2)
- If you have a class method that doesn't use `this`, put it outside of the class. Most of the time those are helper methods which can be put in a separate file
- This will give an error when trying to commit `console.log`
- It will also warn when awaiting a promise in the return statement
- Adds a strict equality check - need to use === (except for null/undefined, where `!= undefined` would match both)
- Warn when there are multiple imports from the same module on different lines (they must be merged)

The new rules are in `tslint.newrules.json` and you can get more informations on this page: https://palantir.github.io/tslint/rules/
